iroh-net/src/hp/netcheck/reportgen.rs

```diff
@@ -310,16 +318,28 @@ impl Actor {
 
         // A collection of futures running probe sets.
         let mut probes = FuturesUnordered::default();
+        let mut derp_nodes_cache: BTreeMap<String, Box<DerpNode>> = BTreeMap::new();
 
-        for probe_set in self.plan.values() {
+        for probe_set in plan.values() {
             let mut set = FuturesUnordered::default();
             for probe in probe_set {
                 let reportstate = self.addr();
                 let stun_sock4 = self.stun_sock4.clone();
                 let stun_sock6 = self.stun_sock6.clone();
-                let node = super::named_node(&self.derpmap, probe.node());
-                ensure!(node.is_some(), "missing named node {}", probe.node());
-                let node = node.unwrap().clone();
+                let derp_node = match derp_nodes_cache.get(probe.node()) {
+                    Some(node) => node.clone(),
+                    None => {
+                        let name = probe.node().to_string();
+                        let node = self
+                            .derp_map
+                            .named_node(&name)
+                            .with_context(|| format!("missing named derp node {}", probe.node()))?;
+                        let node = Box::new(node.clone());
+                        derp_nodes_cache.insert(name, node.clone());
```

Comment:
I am pretty sure that this will clone the full `DerpNode`, If you want ref counting you need to use `Rc` or `Arc`.

Reply:
ignore, this was on an old commit